### PR TITLE
Features/display roadmap 80

### PR DESCRIPTION
**Pull request by mballeza**
Still needs an updated first marker, Legend text, has some leftover comments, etc. However it would be nice to get some feedback so I can finish up tomorrow.

### CODE REVIEW

**Comment by mballeza** [comment]
Okay this is a larger pull request, so here goes. There are several changes summarized below:
- Routing
  - Added a route for "/roadmap/:id". This was the only approach that made differentiating which profile to load onto the map component easier for me.
- Common assets
  - Moved the icons to assets/icons folder.
  - Moved getDate function to constants/dates folder.
  - Copied over the media reducer function to the User Reducer.
- Various styling adjustments.
- Profile loading
  - While signed-in:
    - Home page should only show current user roadmap (using User Reducer).
  - While signed-out:
    - Home page should render a random user's roadmap (using ProfessionalPage Reducer).

**Comment by mballeza** [comment]
Changes made from my last comment:
- Profile loading
  - Upon sign-in
    - Routes to current user's roadmap. The key thing to take note is that after signing in, the user has to navigate to their profile page and then click on "View Map" or enter the url manually.
  - While signed-in
    - MESA Home page button will load a random roadmap.
  - While signed-out
    - MESA Home page button will load a random roadmap.

Please try out these scenarios to check to make sure I didn't miss anything bizarre. Thanks!
1. Sign-in: should route you to your roadmap.
2. MESA Home page button: should always route to home page with random roadmap.

**Comment by djarredon** [comment]
- I really like the addition of the user's profile picture in the navbar when signed in, but can you add `<Link to={'/profile'}>...</Link>` around it to link it to the current user's profile.
- I'm getting a 404 error when navigating to '/profile'

**Review by Domicilius — APPROVED** [review]

File: client/src/components/map/style.css [hunk 1]
```diff
@@ -157,6 +154,10 @@
     height: calc(100vh - 56px - 100px);
   }
 
+  .mapProfilePreviewInfo {
+    margin-left: 0px;
```
Comment (halalak) [inline client/src/components/map/style.css]: is it necessary to have a margin-left if you're applying 0px?

Comment (mballeza) [reply client/src/components/map/style.css]: Yeah, it overrides the margin-left above (line 58).

**Review by djarredon — APPROVED** [review]

File: client/src/components/map/style.css (hunk 1 above)
Comment (halalak) [reply client/src/components/map/style.css]: ah, gotcha

**Review by halalak — APPROVED** [review]

**Pull request merged**